syncopy/tests/test_selectdata.py

```diff
@@ -9,749 +9,555 @@
-
-        # check original shapes
-        assert all([trl.shape[1] == self.nChannels for trl in ang.trials])
-        assert all([trl.shape[0] == self.lenTrial for trl in ang.trials])
-
-        # test inplace channel, trial and latency selection
-        # ang.time[0] = array([0.5, 1. , 1.5, 2. , 2.5])
-        # this latency selection hence takes the last two samples
-        select = {'channel': [2, 7, 9], 'trials': [0, 3, 5], 'latency': [1, 2]}
-        ang.selectdata(**select, inplace=True)
-
-        # now check shapes and number of trials returned by Selector
-        # checks channel axis
-        assert all([trl.shape[1] == 3 for trl in ang.selection.trials])
-        # checks time axis
-        assert len(ang.selection.trials) == 3
-
-        # test for non-existing trials, trial indices are relative here!
-        select = {'trials': [0, 3, 5]}
-        ang.selectdata(**select, inplace=True)
-        assert ang.selection.trial_ids[2] == 5
-        # this returns original trial 6 (with index 5)
-        assert np.array_equal(ang.selection.trials[2], ang.trials[5])
-        # we only have 3 trials selected here, so max. relative index is 2
-        with pytest.raises(SPYValueError, match='less or equals 2'):
-            ang.selection.trials[5]
-
-        # Fancy indexing is not allowed so far
-        select = {'channel': [7, 7, 8]}
-        ang.selectdata(**select, inplace=True)
-        with pytest.raises(SPYValueError, match='fancy selection with repetition'):
-            ang.selection.trials[0]
-        select = {'channel': [7, 3, 8]}
-        ang.selectdata(**select, inplace=True)
-        with pytest.raises(SPYValueError, match='fancy non-ordered selection'):
-            ang.selection.trials[0]
-
-    def test_parallel(self, testcluster):
+    trldef = np.vstack([np.arange(0, nSamples * nTrials, nSamples),
+                        np.arange(0, nSamples * nTrials, nSamples) + nSamples,
+                        np.ones(nTrials) * -1]).T
+
+    # this is an array running from 1 - nChannels * nSamples * nTrials
+    # with shape: nSamples*nTrials x nChannels
+    # and with data[i, j] = i+1 + j * nSamples*nTrials
+    data = np.arange(1, nTrials * nChannels * nSamples + 1).reshape(nChannels, nSamples * nTrials).T
+
+    adata = spy.AnalogData(data=data, samplerate=samplerate,
+                           trialdefinition=trldef)
+
+    def test_ad_selection(self):
+
+        """
+        Create a typical selection and check that the returned data is correct
+        """
+
+        selection = {'trials': 1, 'channel': [6, 2], 'latency': [0, 1]}
+        res = spy.selectdata(self.adata, selection)
+
+        # pick the data by hand, latency [0, 1] covers 2nd - 4th sample index
+        # as time axis is array([-0.5,  0. ,  0.5,  1. ,  1.5])
+
+        solution = self.adata.data[self.nSamples:self.nSamples * 2]
+        solution = np.column_stack([solution[1:4, 6], solution[1:4, 2]])
```

Comment:
I found this a bit cryptic at first because the comment only talks about the latency selction but the trials and channels are being selected here too.

Reply:
Yes, as it's easy to grasp what are the availabel channels and trials, yet for the latency it is not that obvious!